packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -431,6 +431,38 @@ describe("Runtime", () => {
 					})),
 				);
 			});
+
+			it("Can't call flush() inside ensureNoDataModelChanges's callback", async () => {
+				containerRuntime = await ContainerRuntime.load(
+					getMockContext() as IContainerContext,
+					[],
+					undefined, // requestHandler
+					{}, // runtimeOptions
+				);
+
+				assert.throws(() =>
+					containerRuntime.ensureNoDataModelChanges(() => {
+						(containerRuntime as any).flush();
```

Comment:
Hard to say what it tests - why should we care what happens if someone disrespects boundaries? :)

Reply:
This tests there's an error when you call flush. How flush is called is an implementation detail and this is the easiest way to do it in this unit test. This respects boundaries https://github.com/microsoft/FluidFramework/pull/16008/files#diff-562098eab98c8630e03564af13e114fb1f62acc6ca629b3fb1a255618131db09R275 so I'll change this test to something less unorthodox.